### ISSUE TITLE
ToString instance for Xml does not correctly escape

### ISSUE THREAD

**Issue by eric-wieser** [leanprover/lean4#4411]
### Prerequisites

Please put an X between the brackets as you perform the following steps:

* [x] Check that your issue is not already filed:
      https://github.com/leanprover/lean4/issues
* [x] Reduce the issue to a minimal, self-contained, reproducible test case.
      Avoid dependencies to Mathlib or Batteries.
* [x] Test your test case against the latest nightly release, for example on
      https://live.lean-lang.org/#project=lean-nightly
      (You can also use the settings there to switch to “Lean nightly”)

### Description

While the XML parser correctly handles XML escape sequences, the stringifier does not emit them.

### Context

[Broader context that the issue occured in. If there was any prior discussion on [the Lean Zulip](https://leanprover.zulipchat.com), link it here as well.]

### Steps to Reproduce

Run the following:
```lean4
import Lean

run_cmd do
  let .ok doc := Lean.Xml.parse "<b class=\"&quot;\">hello &lt; world </b>" | unreachable!
  Lean.logInfo (toString doc)
```

**Expected behavior:** The output is `<b class="&quot;">hello &lt; world </b>`

**Actual behavior:** The output is `<b class=""">hello < world </b>`, which is invalid HTML

### Versions

4.9.0-rc1  
4.10.0-nightly-2024-06-09

### Additional Information

[Additional information, configuration or data that might be necessary to reproduce the issue]

### Impact

Add :+1: to [issues you consider important](https://github.com/leanprover/lean4/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc). If others are impacted by this issue, please ask them to add :+1: to it.

**Comment by acmepjz** [comment]
I'd like to open a PR to fix this. Any suggestions?

**Comment by keithadler** [comment]
This looks obsolete rather than fixed: `Lean.Data.Xml` was removed from core in #12302 (`chore: remove unused Lean.Data.Xml`, 2026-02-04).

On `4.35.0-nightly-2026-09-04` the repro now fails with `Unknown identifier 'Lean.Xml.parse'`, and there is no `src/Lean/Data/Xml` in the tree.

Suggest closing as obsolete.

Triage assisted by Claude; checked manually.

**Comment by acmepjz** [comment]
> `Lean.Data.Xml` was removed

Yes you are right; it was originally used by doc-gen4 and now is migrated to https://github.com/dupuisf/BibtexQuery/blob/master/BibtexQuery/Xml.lean. Before closing this issue, maybe we should check if it's fixed in there.
